src/frontend/src/handler/create_table.rs

```diff
@@ -1709,6 +1714,82 @@ pub async fn create_iceberg_engine_table(
 
     sink_with.insert("is_exactly_once".to_owned(), "true".to_owned());
 
+    let partition_by = handler_args
+        .with_options
+        .get("partition_by")
+        .map(|v| v.to_owned());
+
+    if let Some(partition_by) = &partition_by {
+        // captures column, transform(column), transform(n,column), transform(n, column)
+        let re =
+            Regex::new(r"(?<transform>\w+)(\(((?<n>\d+)?(?:,|(,\s)))?(?<field>\w+)\))?").unwrap();
+        if !re.is_match(partition_by) {
+            bail!(format!(
+                "Invalid partition fields: {}\nHINT: Supported formats are column, transform(column), transform(n,column), transform(n, column)",
+                partition_by
+            ))
+        }
```

Comment:
These code are copy-pasted from sink? 🤔

Reply:
Not sure whether we need keep both of them. If needed, maybe better to have a common `fn(String) ->  Vec<UnboundPartitionField>` instead